src/adapter/src/catalog.rs

```diff
@@ -5172,19 +5180,32 @@ impl Catalog {
                 computectl_addrs,
                 compute_addrs,
                 workers,
-            } => ReplicaLocation::Unmanaged(UnmanagedReplicaLocation {
-                storagectl_addrs,
-                storage_addrs,
-                computectl_addrs,
-                compute_addrs,
-                workers,
-            }),
+            } => {
+                if !allowed_availability_zones.is_none() {
```

Comment:
the double negative feels weird `not none` vs `some` 🤷‍♂️

Reply:
you are right!